config/default.yml

```
@@ -813,6 +813,16 @@ Rails/SquishedSQLHeredocs:
   # to be preserved in order to work, thus auto-correction is not safe.
   SafeAutoCorrect: false
 
+Rails/TableNameAssignment:
+  Description: >-
+    Do not use `self.table_name =`. Use Inflections or `table_name_prefix` instead.
+  StyleGuide: 'https://rails.rubystyle.guide/#keep-ar-defaults'
+  Enabled: false
+  VersionAdded: '<<next>>'
+  Exclude:
+    - '**/spec/**/*'
+    - db/migrate/*.rb
```

I think the following is sufficient because the cop targets Active Record. How about?

```suggestion
  Include:
    - app/models/**/*.rb
```

---

Maybe `**/app/models/**/*.rb` to also cover:
 - api/app/models/spree/oauth_access_grant.rb (spree)
 - core/app/models/spree/store_product.rb (spree)

However, Spree seems to be the only project that adheres to in-project engine/gem subdirectory.
I'm totally fine with the suggested `app/models/**/*.rb`, too.

---

Yep. I think `app/models/**/*.rb` supported by the Rails standard (`rails new`) would be better. The extended path will be customizable by user :-)